website/content/specification/syntax/constraints.md

```diff
@@ -179,6 +179,14 @@ A constraint may have an OPTIONAL [`@level`](#level) attribute and/or an OPTIONA
 
 If defined, the `<message>` value MUST be a [Metaschema string value](/specification/datatypes#string). It MAY contain a Metapath expression templates that starts with `{`, contains a Metapath expression, and ends with `}`.  When evaluating a template Metapath expression, the context of the Metapath [evaluation focus](#constraint-processing) MUST be the failing value node.
 
+##`matches`
+
+The `matches` constraint is a type of Metaschema constraint that restricts field or flag value(s) based on a component matching the target Metapath expression.
+
+The `@target` attribute of an `<matches>` constraint specifies the node in a component definition whose value is equal. The `matches` constraint MUST define a [`target`](#target) with a Metapath expression.
+
```

Comment:
Also discussed in Element, can we please include information about these syntax elements, specifically a) if they are required or optional, b) what is their purpose and c) match additional details with the above elements?

- [ ] matches/@datatype
- [ ] key-field/@pattern
- [ ] key-field/@regex

Reply:
I edited the correction above. Arminta caught me in a silly lie because I did not review this comment _while_ using the Metaschema XML schema and/or testing a document instance, removing `key-field/@pattern` and properly pointing out `@regex` is on `matches`, this type has no `key-field` child element!

Reply:
Resolved during pairing session.